src/Microsoft.Health.Fhir.Shared.Api/Modules/FeatureFlags/ConditionalUpdates/ConditionalUpsertFeatureModule.cs

```diff
@@ -0,0 +1,29 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Api.Configs;
+using Microsoft.Health.Fhir.Core.Features.Resources.Upsert;
+
+namespace Microsoft.Health.Fhir.Api.Modules.FeatureFlags.ConditionalUpdates
+{
+    public class ConditionalUpsertFeatureModule : IStartupModule
+    {
+        public void Load(IServiceCollection services)
+        {
+            services.Add<ConditionalUpsertResourceHandler.IsEnabled>(x => () => x.GetRequiredService<IOptions<FeatureConfiguration>>().Value.SupportsConditionalUpdate)
```

Comment:
This is nice, but I wonder if it would be simpler for the constructor to just take a FeatureConfiguration

Reply:
We could. The next question is does it make sense to move Configuration into Core? This was the original constraint for this

Reply:
Yeah ok, probably more modular this way.